moabb/tests/test_splits.py

```diff
@@ -74,13 +107,158 @@ def test_is_shuffling():
+    _, y, metadata = data
+
+    splitter = CrossSessionSplitter(shuffle=shuffle, random_state=random_state)
+
+    for (idx_train, idx_test), (idx_train_splitter, idx_test_splitter) in zip(
+        eval_split_cross_session(shuffle=shuffle, random_state=random_state, data=data),
+        splitter.split(y, metadata),
+    ):
+        assert np.array_equal(idx_train, idx_train_splitter)
+        assert np.array_equal(idx_test, idx_test_splitter)
+
+
+def test_cross_session_is_shuffling_and_order(data):
+    _, y, metadata = data
+
+    splitter_no_shuffle = CrossSessionSplitter(shuffle=False)
+    splitter_shuffle = CrossSessionSplitter(
+        shuffle=True, random_state=3, cv_class=GroupShuffleSplit
+    )
+
+    splits_no_shuffle = list(splitter_no_shuffle.split(y, metadata))
+    splits_shuffle = list(splitter_shuffle.split(y, metadata))
+
+    train_diff = []
+    test_diff = []
+
+    # For tracking session order differences
+    session_orders_no_shuffle = []
+    session_orders_shuffle = []
+
+    for i, ((train_ns, test_ns), (train_s, test_s)) in enumerate(
+        zip(splits_no_shuffle, splits_shuffle)
+    ):
+        print(f"\nFold {i}:")
+
+        # Get session ordering for non-shuffled and shuffled
+        train_ns_sessions = metadata.iloc[train_ns]["session"].unique()
+        test_ns_sessions = metadata.iloc[test_ns]["session"].unique()
+        train_s_sessions = metadata.iloc[train_s]["session"].unique()
+        test_s_sessions = metadata.iloc[test_s]["session"].unique()
+
+        print(f"Train no shuffle sessions: {train_ns_sessions}")
+        print(f"Test no shuffle sessions : {test_ns_sessions}")
+        print(f"Train shuffled sessions  : {train_s_sessions}")
+        print(f"Test shuffle sessions    : {test_s_sessions}")
+
+        # Track if indices are the same
+        train_diff.append(np.array_equal(train_ns, train_s))
+        test_diff.append(np.array_equal(test_ns, test_s))
+
+        # Track session orders
+        session_orders_no_shuffle.append(
+            (list(train_ns_sessions), list(test_ns_sessions))
+        )
+        session_orders_shuffle.append((list(train_s_sessions), list(test_s_sessions)))
+
+    # Check if indices are different in at least some folds
+    assert not all(train_diff), "All train indices are identical despite shuffle"
+    assert not all(test_diff), "All test indices are identical despite shuffle"
+
+    # Check if session ordering is different
+    session_order_differences = [
+        not (
+            np.array_equal(no_shuffle[0], shuffle[0])
+            and np.array_equal(no_shuffle[1], shuffle[1])
+        )
+        for no_shuffle, shuffle in zip(session_orders_no_shuffle, session_orders_shuffle)
+    ]
+
+    assert any(session_order_differences), (
+        "Session ordering is identical in all folds despite shuffle. "
+        "When shuffle=True, we expect some difference in the session ordering."
+    )
```

Comment:
I don't think if this test is supposed to check my previous remark but my question was: are the sessions for `subject1` not split the same as for `subject2`.

Reply:
I created one test. Can you check it?